src/contracts/permissions/PermissionController.sol

```diff
@@ -0,0 +1,176 @@
+
+        emit DelegateSet(account, delegate, target, selector);
+    }
+
+    /// @inheritdoc IPermissionController
+    function removeDelegate(
+        address account,
+        address delegate,
+        address target,
+        bytes4 selector
+    ) external onlyAdmin(account) {
+        AccountPermissions storage permissions = _permissions[account];
+
+        bytes32 targetSelector = _encodeTargetSelector(target, selector);
+        require(permissions.delegatePermissions[delegate].contains(targetSelector), DelegateNotSet());
+
+        // Remove the delegate from the account's permissions
+        permissions.delegatePermissions[delegate].remove(targetSelector);
+        permissions.permissionDelegates[targetSelector].remove(delegate);
+
+        emit DelegateRemoved(account, delegate, target, selector);
+    }
+
+    /**
+     *
+     *                         INTERNAL FUNCTIONS
+     *
+     */
+
+    /// @dev Encodes the target and selector into a single bytes32 value
+    function _encodeTargetSelector(address target, bytes4 selector) internal pure returns (bytes32) {
+        return bytes32(abi.encodePacked(target, uint96(bytes12((selector)))));
+    }
+
+    /// @dev Decodes the target and selector from a single bytes32 value
+    function _decodeTargetSelector(
+        bytes32 targetSelector
+    ) internal view returns (address, bytes4) {
+        address target = address(uint160(uint256(targetSelector) >> 96));
+        // The selector is in the lower 32 bits of the targetSelector
+        bytes4 selector = bytes4(uint32(uint256(targetSelector) >> 64));
+
+        return (target, selector);
+    }
+
+    /**
+     *
+     *                         VIEW FUNCTIONS
+     *
+     */
+
+    /// @inheritdoc IPermissionController
+    function isAdmin(address account, address caller) public view returns (bool) {
+        if (_permissions[account].admins.length() == 0) {
+            // If the account does not have an admin, the caller must be the account
+            return account == caller;
+        } else {
+            // If the account has an admin, the caller must be an admin
+            return _permissions[account].admins.contains(caller);
+        }
+    }
+
+    /// @inheritdoc IPermissionController
+    function getAdmins(
+        address account
+    ) external view returns (address[] memory) {
+        if (_permissions[account].admins.length() == 0) {
+            address[] memory admin = new address[](1);
+            admin[0] = account;
+            return admin;
+        } else {
+            return _permissions[account].admins.values();
+        }
+    }
+
+    /// @inheritdoc IPermissionController
+    function canCall(address account, address caller, address target, bytes4 selector) external view returns (bool) {
+        return isAdmin(account, caller)
+            || _permissions[account].delegatePermissions[caller].contains(_encodeTargetSelector(target, selector));
```

Comment:
Can we make a view function

`isPermissioned(address account, address caller, address target, bytes4 selector);`

that does the logic after `||`

Reply:
Not sure why we would want to call a subset of this function?